physics/eclipse_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include "geometry/epoch.hpp"
+#include "geometry/named_quantities.hpp"
+#include "geometry/grassmann.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
+#include "physics/solar_system.hpp"
+#include "quantities/elementary_functions.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+#include "testing_utilities/numerics.hpp"
+
+namespace principia {
+
+using astronomy::ICRFJ2000Equator;
+using geometry::JulianDate;
+using geometry::Sign;
+using integrators::McLachlanAtela1992Order5Optimal;
+using quantities::ArcCos;
+using quantities::si::Kilo;
+using quantities::si::Metre;
+using quantities::si::Milli;
+using quantities::si::Minute;
+using quantities::si::Nano;
+using testing_utilities::AbsoluteError;
+using ::testing::AllOf;
+using ::testing::Gt;
+using ::testing::Lt;
+
+namespace physics {
+
+class EclipseTest : public testing::Test {
+ protected:
+  EclipseTest() {
+    solar_system_1950_.Initialize(
+        SOLUTION_DIR / "astronomy" / "gravity_model.proto.txt",
+        SOLUTION_DIR / "astronomy" /
+            "initial_state_jd_2433282_500000000.proto.txt");
+  }
+
+  void CheckLunarEclipse(Instant const& current_time, Sign const moon_offset) {
+    auto ephemeris = solar_system_1950_.MakeEphemeris(
+        McLachlanAtela1992Order5Optimal<Position<ICRFJ2000Equator>>(),
+        45 * Minute, 5 * Milli(Metre));
+
+    ephemeris->Prolong(
+        JulianDate(2433374.5));  // Prolong just past date of eclipse
+                                 // (Eclipse was 1950-04-02 but JD
+                                 // is 1950-04-03 00:00:00).
+    auto const sun = solar_system_1950_.massive_body(*ephemeris, "Sun");
+    auto const earth = solar_system_1950_.massive_body(*ephemeris, "Earth");
+    auto const moon = solar_system_1950_.massive_body(*ephemeris, "Moon");
+
+    auto const q_sun = ephemeris->trajectory(sun)->EvaluatePosition(
+        current_time, /* hint= */ nullptr);
```

Comment:
No spaces within the comment, nor after it, thus
```cpp
/*hint=*/nullptr
```
We are currently not consistently following this convention throughout the code (sometimes we use `arg  // arg`, sometimes `arg /*arg*/`), but this new syntax is [checked by clang-tidy](http://clang.llvm.org/extra/clang-tidy/checks/misc-argument-comment.html) (which we might start using at some point).